projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -0,0 +1,165 @@
+				comment.body.includes( '**Support References**' )
+			) {
+				commentID = comment.id;
+			}
+		} );
+	}
+
+	return commentID;
+}
+
+/**
+ * Scan the contents of the issue as well as all its comments, get all support references, and add them to an array of references.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - Issue number.
+ * @returns {Promise<Array>} Promise resolving to an array.
+ */
+async function getIssueReferences( octokit, owner, repo, number ) {
+	const ticketReferences = [];
+	const supportIds = [];
+	const referencesRegexP = /[0-9]*-(?:chat|hc|zen|zd)/gim;
+
+	debug( `gather-support-references: Getting references from comments.` );
+	for await ( const response of octokit.paginate.iterator( octokit.rest.issues.listComments, {
+		owner: owner.login,
+		repo,
+		issue_number: +number,
+	} ) ) {
+		response.data.map( comment => {
+			ticketReferences.push( ...comment.body.matchAll( referencesRegexP ) );
+		} );
+	}
+
+	debug( `gather-support-references: Getting references from issue body.` );
+	const {
+		data: { body },
+	} = await octokit.rest.issues.get( {
+		owner: owner.login,
+		repo,
+		issue_number: +number,
+	} );
+	ticketReferences.push( ...body.matchAll( referencesRegexP ) );
+
+	// Buid a first array with only the support IDs we've collected.
+	ticketReferences.map( reference => {
+		supportIds.push( reference[ 0 ] );
+	} );
+
+	// That array can still include duplicates, or references that are not formatted quite properly.
+	// Let's build a final array with unique and correct support IDs.
+	const correctedSupportIds = [];
+	supportIds.map( supportId => {
+		// xxx-zen and xxx-hc are the preferred formats for tickets and chats.
+		// xxx-zd and xxx-chat, as well as uppercase versions, are considered as alternate versions.
+		const wrongId = supportId.match( /([0-9]*)-(zd|chat)/i );
+		if ( wrongId ) {
+			const correctedId = `${ wrongId[ 1 ] }-${ wrongId[ 2 ] === 'zd' ? 'zen' : 'hc' }`;
+			if ( ! correctedSupportIds.includes( correctedId ) ) {
+				correctedSupportIds.push( correctedId );
+			}
+		} else if ( ! correctedSupportIds.includes( supportId.toLowerCase() ) ) {
+			correctedSupportIds.push( supportId.toLowerCase() );
+		}
+	} );
+
+	return correctedSupportIds;
+}
+
+/**
+ * Creates or updates a comment on issue.
+ *
+ * @param {WebhookPayloadIssue} payload - Issue event payload.
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {Array} issueReferences - Array of support references.
+ */
+async function createOrUpdateComment( payload, octokit, issueReferences ) {
+	const { issue, repository } = payload;
+	const { number } = issue;
+	const { name: repo, owner } = repository;
+	const ownerLogin = owner.login;
+
+	const comment = `**Support References**
+	${ issueReferences
+		.map(
+			reference => `
+- [ ] ${ reference }`
```

Comment:
So every time the bot updates its comment, it'll uncheck all the checkboxes. That doesn't seem like the behavior we want.

Reply:
From what I know, HEs need the checkboxes so they can get back to the users once the issue is closed, and keep track of the users they've contacted.

With that in mind, and since the action is not triggered when issue comments are edited, I don't think it should get triggered once the issue has been closed.

That said, maybe I should add an additional safeguard anyway, so a new comment is created when comments are left once the issue is opened? I'm not sure that's the best way to go about this.

Reply:
I took a stab at this in df0d52c64c1f9f44924ab9d4213252330d768392 (and 20e0387c6a51e890d184975432a757c218110823)